runtime/devnet/src/extensions.rs

```diff
@@ -49,22 +54,12 @@ where
 	fn call<E: Ext>(&mut self, env: Environment<E, InitState>) -> Result<RetVal, DispatchError>
 	where
 		E: Ext<T = T>,
-		T::AccountId: UncheckedFrom<T::Hash> + AsRef<[u8]>,
 	{
 		log::debug!(target:LOG_TARGET, " extension called ");
 		match v0::FuncId::try_from(env.func_id())? {
-			v0::FuncId::Dispatch => {
-				match dispatch::<T, E>(env) {
-					Ok(()) => Ok(RetVal::Converging(0)),
-					Err(DispatchError::Module(error)) => {
-						// encode status code = pallet index in runtime + error index, allowing for
-						// 999 errors
-						Ok(RetVal::Converging(
-							(error.index as u32 * 1_000) + u32::from_le_bytes(error.error),
-						))
-					},
-					Err(e) => Err(e),
-				}
+			v0::FuncId::Dispatch => match dispatch::<T, E>(env) {
+				Ok(()) => Ok(RetVal::Converging(0)),
+				Err(e) => Ok(RetVal::Converging(convert_to_status_code(e))),
```

Comment:
Would prefer `e.into()`

Reply:
`Only traits defined in the current crate can be implemented for arbitrary types [E0117].`

Work around is to create a wrapper for `u32` or `DispatchError`. Perhaps the unsolved problem for the `Other` error return in our chain extension will require a wrapper around DispatchError any way.